include/engine/hdf5/HDF5WriterP.h

```diff
@@ -0,0 +1,224 @@
+/*
+ * HDF5WriterP.h
+ *
+ *  Created on: March 20, 2017
+ *      Author: Junmin
+ */
+
+#ifndef HDF5_WRITER_P_H_
+#define HDF5_WRITER_P_H_
+
+#include <iostream> //std::cout must be removed, only used for hello example
+#include <unistd.h> //sleep must be removed
```

Comment:
unistd isn't used in the header.  Please remove.  If it's still needed in the cpp file then you can include it there.